projects/GKCore/GKCore/Charts/TreeChartModel.cs

```diff
@@ -1014,6 +1033,55 @@ private void RecalcDesc(TreeChartPerson person, ExtPoint aPt, bool predef)
             if (person.Sex == GEDCOMSex.svNone || person.Sex == GEDCOMSex.svUndetermined) {
                 fEdges[gen] = person.Rect.Right;
             }
+
+            // Kevin Routley (aka fire-eggs, aka KBR). Partially address issue #147.
+            // As a result of combining duplicate people together, there may be overlapping
+            // boxes. Here, these overlaps are corrected. NOTE: duplicate people _across_generations_
+            // have not been combined as no adjustment is being made to shift a box down as 
+            // would be necessary.
+            FixOverlap(person);
+        }
+
+        private void AdjustChildren(TreeChartPerson person, int offset)
+        {
+            // Kevin Routley (aka fire-eggs, aka KBR). Partially address issue #147.
+            // a person has been shifted because of collision. Recursively adjust 
+            // _descendants_ of that person
+            if (person.GetChildsCount() < 1)
+                return;
+            int cMax = person.GetChildsCount();
+            for (int i = 0; i < cMax; i++)
+            {
+                var pp = person.GetChild(i);
+                pp.PtX += offset;
+                AdjustChildren(pp, offset);
+            }
+        }
+
+        private int FixOverlap(TreeChartPerson person)
+        {
+            // Kevin Routley (aka fire-eggs, aka KBR). Partially address issue #147.
+            // Check for, and correct, collisions of boxes in this person's 
+            // generational row.
+            int shiftAmount = 0;
+            for (int m = 0; m <= fPersons.Count; m++)

```

Comment:
This is not a mistake by accident?
`m <= fPersons.Count`

Reply:
Yup, it is. Good eye! [There is a loop short-circuit which appears to have prevented it from being an issue in during my testing.]